### PR TITLE
feat: item login users are different than members 

### PR DESCRIPTION
**Pull request by Defferrard**
After this migration, we will keep some orphan account.
Orphan account are created after all Item Login Schema of an account has been deleted.

After the migration, we can delete all orphan account with a SQL query.
> [!WARNING]  
> Please, be careful when running this query. We need to be sure we're not gonna delete a valid Member account.
```sql
DELETE FROM account WHERE email ~* '^[0-9a-f]{4}-[0-9]{10,}@graasp.org' AND type = 'individual'
```

# API Breaking changes
- **GET /mentions** Response items will have an account property instead of member property.
- **PATCH /mentions/:mentionId** Response will have an account property instead of member property.
- **DELETE /mentions/:mentionId** Response will have an account property instead of member property.
- **DELETE /mentions** Response items will have an account property instead of member property.

- **PATCH /items/:itemId/app-data/:id** Response will have an account property instead of member property.
- **POST /items/:itemId/app-data** 
  - Request Body property memberId has been renamed to accountId
  - Response will have an account property instead of member property.
- **DELETE /items/:itemId/app-data/:id** Response will have an account property instead of member property.
- **GET /items/:itemId/app-data** Response will have an account property instead of member property.
- **GET /items/app-data** Response items will have an account property instead of member property.

- **POST /item-memberships**  
  - Request Body property memberId has been renamed to accountId
  - Response will have an account property instead of member property.
- **POST /item-memberships/:itemId** Request Body Items property memberId has been renamed to accountId
- **PATCH /item-memberships/:id** Response will have an account property instead of member property.
- **DELETE /item-memberships/:id** Response will have an account property instead of member property.
- **GET /item-memberships** Response items will have an account property instead of member property.

- All other chatMention object will have an account property instead of member property.
- All other appData object will have an account property instead of member property.
- All other itemMembership object will have an account property instead of member property.
# SDK Breaking changes
- Action member property as been renamed to account
- AuthTokenSubject memberId property as been renamed to account
- AppAction member property as been renamed and typed to account
- AppData member property as been renamed and typed to account
- ChatMessage creator property as been typed to Account
- ChatMention member property as been renamed and typed to account
- ItemLogin type has been removed
- ItemMembership member property as been renamed and typed to account
- PSEUDO_USER_MAIL_REGEX has been removed
- New GuestFactory and baseAccountFactory
- New MemberType Guest
- New types Account, BaseAccount, CompleteAccount, CompleteMember and CompleteGuest

### CODE REVIEW

File: src/services/item/plugins/itemCategory/index.ts [hunk 1]
```diff
@@ -35,18 +35,18 @@ const plugin: FastifyPluginAsync = async (fastify) => {
...
   // insert item category
   fastify.post<{ Params: { itemId: string }; Body: { categoryId: string } }>(
     '/:itemId/categories',
-    { schema: create, preHandler: [isAuthenticated, matchOne(validatedMember)] },
+    { schema: create, preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)] },
     async ({ user, params: { itemId }, body: { categoryId } }) => {
       return db.transaction(async (manager) => {
         return itemCategoryService.post(
-          user?.member,
+          user?.account,
```
Comment (pyphilia) [inline src/services/item/plugins/itemCategory/index.ts:49]: assertIsMember?

File: src/services/item/plugins/itemCategory/index.ts [hunk 2]
```diff
@@ -58,11 +58,11 @@ const plugin: FastifyPluginAsync = async (fastify) => {
   // delete item category entry
   fastify.delete<{ Params: { itemCategoryId: string; itemId: string } }>(
     '/:itemId/categories/:itemCategoryId',
-    { schema: deleteOne, preHandler: [isAuthenticated, matchOne(validatedMember)] },
+    { schema: deleteOne, preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)] },
     async ({ user, params: { itemCategoryId, itemId } }) => {
       return db.transaction(async (manager) => {
         return itemCategoryService.delete(
-          user?.member,
+          user?.account,
```
Comment (pyphilia) [inline src/services/item/plugins/itemCategory/index.ts:65]: assertIsMember?

File: src/services/item/plugins/itemFavorite/index.ts [hunk 3]
```diff
@@ -21,31 +21,31 @@ const plugin: FastifyPluginAsync = async (fastify) => {
     '/favorite',
     { schema: getFavorite, preHandler: isAuthenticated },
     async ({ user }) => {
-      const member = notUndefined(user?.member);
-      return favoriteService.getOwn(member, buildRepositories());
+      const account = notUndefined(user?.account);
+      return favoriteService.getOwn(account, buildRepositories());
```
Comment (pyphilia) [inline src/services/item/plugins/itemFavorite/index.ts:25]: Member only.

File: src/services/item/plugins/itemFavorite/services/favorite.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export class FavoriteService {
     this.itemService = itemService;
   }
 
-  async getOwn(actor: Member, repositories: Repositories): Promise<PackedItemFavorite[]> {
+  async getOwn(actor: Account, repositories: Repositories): Promise<PackedItemFavorite[]> {
```
Comment (pyphilia) [inline src/services/item/plugins/itemFavorite/services/favorite.ts:19]: Member?

File: src/services/item/plugins/itemFavorite/services/favorite.ts [hunk 5]
```diff
@@ -40,15 +40,15 @@ export class FavoriteService {
     });
   }
 
-  async post(actor: Member, repositories: Repositories, itemId: string) {
+  async post(actor: Account, repositories: Repositories, itemId: string) {
```
Comment (pyphilia) [inline src/services/item/plugins/itemFavorite/services/favorite.ts:43]: Member?

File: src/services/item/plugins/itemFavorite/services/favorite.ts [hunk 6]
```diff
@@ -40,15 +40,15 @@ export class FavoriteService {
...
     const { itemFavoriteRepository } = repositories;
 
     // get and check permissions
     const item = await this.itemService.get(actor, repositories, itemId, PermissionLevel.Read);
     return itemFavoriteRepository.post(item.id, actor.id);
   }
 
-  async delete(actor: Member, repositories: Repositories, itemId: string) {
+  async delete(actor: Account, repositories: Repositories, itemId: string) {
```
Comment (pyphilia) [inline src/services/item/plugins/itemFavorite/services/favorite.ts:51]: Member?

File: src/services/item/plugins/itemFlag/index.ts [hunk 7]
```diff
@@ -23,17 +24,18 @@ const plugin: FastifyPluginAsync = async (fastify) => {
...
   );
 
   // create item flag
   fastify.post<{ Params: { itemId: string }; Body: Partial<ItemFlag> }>(
     '/:itemId/flags',
-    { schema: create, preHandler: [isAuthenticated, matchOne(validatedMember)] },
+    { schema: create, preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)] },
     async ({ user, params: { itemId }, body }) => {
+      const member = notUndefined(user?.account);
+      assertIsMember(member);
```
Comment (pyphilia) [inline src/services/item/plugins/itemFlag/index.ts:37]: I think guest can flag items.

File: src/services/item/plugins/itemLike/index.ts [hunk 8]
```diff
@@ -27,8 +28,8 @@ const plugin: FastifyPluginAsync = async (fastify) => {
     '/liked',
     { schema: getLikesForMember, preHandler: isAuthenticated },
     async ({ user }) => {
-      const member = notUndefined(user?.member);
-      return itemLikeService.getForMember(member, buildRepositories());
+      const account = notUndefined(user?.account);
+      return itemLikeService.getForMember(account, buildRepositories());
```
Comment (pyphilia) [inline src/services/item/plugins/itemLike/index.ts:32]: If only validated members can add likes, then only members can get likes.

File: src/services/item/plugins/itemLike/service.ts [hunk 9]
```diff
@@ -10,7 +11,7 @@ export class ItemLikeService {
     this.itemService = itemService;
   }
 
-  async getForMember(actor: Member, repositories: Repositories) {
+  async getForMember(actor: Account, repositories: Repositories) {
```
Comment (pyphilia) [inline src/services/item/plugins/itemLike/service.ts:14]: Rename? Member only?

File: src/services/item/plugins/itemLike/service.ts [hunk 10]
```diff
@@ -38,16 +39,16 @@ export class ItemLikeService {
...
     // QUESTION: allow public to be liked?
-    const item = await this.itemService.get(actor, repositories, itemId);
+    const item = await this.itemService.get(member, repositories, itemId);
 
-    return itemLikeRepository.deleteOne(actor, item);
+    return itemLikeRepository.deleteOne(member, item);
   }
 
-  async post(actor: Member, repositories: Repositories, itemId: string) {
+  async post(actor: Account, repositories: Repositories, itemId: string) {
```
Comment (pyphilia) [inline src/services/item/plugins/itemLike/service.ts:51]: Member only?

File: src/services/item/plugins/itemTag/index.ts [hunk 11]
```diff
@@ -55,27 +57,29 @@ const plugin: FastifyPluginAsync = async (fastify) => {
...
 
   // delete item tag
   fastify.delete<{ Params: { itemId: string; type: ItemTagType } & IdParam }>(
     '/:itemId/tags/:type',
-    { schema: deleteOne, preHandler: [isAuthenticated, matchOne(validatedMember)] },
+    { schema: deleteOne, preHandler: [isAuthenticated, matchOne(validatedMemberAccountRole)] },
     async ({ user, params: { itemId, type } }) => {
       return db.transaction(async (manager) => {
-        return itemTagService.deleteOne(user?.member, buildRepositories(manager), itemId, type);
+        return itemTagService.deleteOne(user?.account, buildRepositories(manager), itemId, type);
```
Comment (pyphilia) [inline src/services/item/plugins/itemTag/index.ts:82]: assertIsMember?

File: src/services/item/plugins/publication/published/service.ts [hunk 12]
```diff
@@ -193,7 +197,7 @@ export class ItemPublishedService {
     return published;
   }
 
-  async delete(actor: Member, repositories: Repositories, itemId: string) {
+  async delete(actor: Account, repositories: Repositories, itemId: string) {
```
Comment (pyphilia) [inline src/services/item/plugins/publication/published/service.ts:200]: Member?

File: src/services/item/plugins/publication/published/repositories/itemPublished.ts [hunk 13]
```diff
@@ -53,7 +54,7 @@ export class ItemPublishedRepository {
     });
   }
 
-  async getForMember(memberId: Member['id']): Promise<Item[]> {
+  async getForMember(accountId: Account['id']): Promise<Item[]> {
```
Comment (pyphilia) [inline src/services/item/plugins/publication/published/repositories/itemPublished.ts:57]: Rename? Only members can get their own published items.

File: src/services/item/plugins/publication/validation/index.ts [hunk 14]
```diff
@@ -34,7 +35,7 @@ const plugin: FastifyPluginAsync = async (fastify) => {
       preHandler: isAuthenticated,
```
Comment (pyphilia) [inline src/services/item/plugins/publication/validation/index.ts:35]: Validations should be available only to members.

File: src/services/item/plugins/publication/validation/service.ts [hunk 15]
```diff
@@ -54,33 +55,33 @@ export class ItemValidationService {
   }
```
Comment (pyphilia) [inline src/services/item/plugins/publication/validation/service.ts:56]: Same as above.

File: src/services/item/plugins/recycled/index.ts [hunk 16]
```diff
@@ -45,8 +46,8 @@ const plugin: FastifyPluginAsync<RecycledItemDataOptions> = async (fastify, opti
     '/recycled',
     { schema: getRecycledItemDatas, preHandler: isAuthenticated },
     async ({ user }) => {
-      const member = notUndefined(user?.member);
-      const result = await recycleBinService.getAll(member, buildRepositories());
+      const account = notUndefined(user?.account);
+      const result = await recycleBinService.getAll(account, buildRepositories());
```
Comment (pyphilia) [inline src/services/item/plugins/recycled/index.ts:50]: Recycle bin should be available only for members, all endpoints.

File: src/services/item/plugins/recycled/repository.ts [hunk 17]
```diff
@@ -3,13 +3,14 @@ import { In } from 'typeorm';
...
 import { AppDataSource } from '../../../../plugins/datasource';
+import { Account } from '../../../account/entities/account';
 import { Member } from '../../../member/entities/member';
 import { Item } from '../../entities/Item';
 import { RecycledItemData } from './RecycledItemData';
 import { CannotRestoreNonDeletedItem } from './errors';
 
 export const RecycledItemDataRepository = AppDataSource.getRepository(RecycledItemData).extend({
-  async getOwnRecycledItemDatas(member: Member): Promise<RecycledItemData[]> {
+  async getOwnRecycledItemDatas(account: Account): Promise<RecycledItemData[]> {
```
Comment (pyphilia) [inline src/services/item/plugins/recycled/repository.ts:13]: Member?

File: src/services/item/plugins/recycled/service.ts [hunk 18]
```diff
@@ -19,7 +20,7 @@ export class RecycledBinService {
     };
   }>();
 
-  async getAll(actor: Member, repositories: Repositories) {
+  async getAll(actor: Account, repositories: Repositories) {
```
Comment (pyphilia) [inline src/services/item/plugins/recycled/service.ts:23]: Member?

File: src/services/item/plugins/recycled/service.ts [hunk 19]
```diff
@@ -79,7 +80,7 @@ export class RecycledBinService {
     return itemsResult;
   }
 
-  async restoreMany(actor: Member, repositories: Repositories, itemIds: string[]) {
+  async restoreMany(actor: Account, repositories: Repositories, itemIds: string[]) {
```
Comment (pyphilia) [inline src/services/item/plugins/recycled/service.ts:83]: Member?

File: src/services/item/repository.ts [hunk 20]
```diff
@@ -657,17 +658,17 @@ export class ItemRepository {
 
   /**
    * Return published items for given member
-   * @param memberId
+   * @param accountId
    * @returns published items for given member
    */
-  async getPublishedItemsForMember(memberId: string): Promise<Item[]> {
+  async getPublishedItemsForMember(accountId: string): Promise<Item[]> {
```
Comment (pyphilia) [inline src/services/item/repository.ts:664]: MemberId?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Questions:
- So you removed ItemLogin and transformed it in Guest(itemLoginSchema) and MemberPassword(account). Wouldn't it be simpler to have password of the guest in their entities directly? I feel like mixing passwords is not so good.
- App settings can only be created, modified, deleted by a regular member. That's the main difference with app-data, where account can create, modify, delete them.
- I'm not quite sure when you need to use `assertIsMember` or not so you can ignore some comments if it is for typing purpose only.
- For likes I would say guest can like items, so depending on what we decide some of my comments can be discarded.

I'm missing 50 files but I think it's a good start^^

File: src/services/auth/plugins/password/entities/password.ts [hunk 21]
```diff
@@ -10,17 +10,17 @@ import {
...
 export class MemberPassword extends BaseEntity {
   @PrimaryGeneratedColumn('uuid')
   id: string;
 
-  @OneToOne(() => Member, (member) => member.id, { onDelete: 'CASCADE' })
-  @JoinColumn({ name: 'member_id' })
-  member: Member;
+  @OneToOne(() => Account, (account) => account.id, { onDelete: 'CASCADE' })
+  @JoinColumn({ name: 'account_id', foreignKeyConstraintName: 'FK_member_password_account_id' })
+  account: Account;
```
[the remark this answers is not in the document]
Comment (Defferrard) [reply src/services/auth/plugins/password/entities/password.ts:23]: We're gonna create a 2nd table "GuestPassword" with guest id, password, updated at, created at

File: src/services/item/plugins/app/appSetting/service.ts [hunk 22]
```diff
@@ -68,68 +69,58 @@ export class AppSettingService {
...
     });
 
     const appSetting = await appSettingRepository.patch(itemId, appSettingId, body);
-    await this.hooks.runPostHooks('patch', member, repositories, {
+    await this.hooks.runPostHooks('patch', account, repositories, {
       appSetting,
       itemId,
     });
     return appSetting;
   }
```
[the remark this answers is not in the document]
Comment (Defferrard) [reply src/services/item/plugins/app/appSetting/service.ts:85]: Only Validated member for appSettings

File: src/services/item/plugins/publication/published/repositories/itemPublished.ts (hunk 13 above)
Comment (Defferrard) [reply src/services/item/plugins/publication/published/repositories/itemPublished.ts:57]: Is this function only used for tests cases ? I didn't been able to find anything else..

File: src/services/file/service.ts [hunk 23]
```diff
@@ -40,7 +41,7 @@ class FileService {
     return this.repository.getFileSize(filepath);
   }
 
-  async upload(member: Member, data: { file: Readable; filepath: string; mimetype?: string }) {
+  async upload(account: Account, data: { file: Readable; filepath: string; mimetype?: string }) {
```
[the remark this answers is not in the document]
Comment (Defferrard) [reply src/services/file/service.ts:44]: It would break `POST /app-data/upload`, is it okay to restrict this route only to members ?

**Comment by spaenleh** [comment]
I am currently working on the frontend changes for this feature in builder. I see that the changes in #1274 have not been added here yet. We need to coordinate to see how we can move forward with this feature without having to rebase this another 100 times.

**Review by spaenleh — APPROVED** [review]

**Pull request merged**